src/hevm/src/EVM/Stepper.hs

```diff
@@ -57,6 +58,9 @@ data Action a where
   -- | Embed a VM state transformation
   EVM  :: EVM a   -> Action a
 
+  -- | Perform an IO action
+  IOAct :: StateT VM IO a -> Action a -- they should all just be this?
```

Comment:
its more direct, we don't have to define new `Query `s for every side effect we want to do

Reply:
yeah, that's what the comment suggests

Reply:
I'm a little on the fence about this change. In a pre `IOAct` world we had pretty strict control over what kind of side effects can be triggered by EVM execution,  but with `IOAct` we allow unbounded side effects. While IOAct is definately simpler and more flexible, I do think that having tight control over side effects in the core is pretty nice. It also feels a little messy to leave the old `Query` types around side by side with the new `IOAct` approach.

I think for this PR all we really needed was a `pleaseGetRandom` query that calls `generate` on a `Gen` or something like that?

Reply:
To be clear I'm not totally opposed to replacing `Query` with `IOAct`, I just think it's a moderately large shift in architectural philosophy, and I might prefer to make this change in a seperate PR where we also remove `Query` entirely.

Reply:
Making a `GetRandom` `Query` here feels a little bureaucratic to me, since these calldata generation calls are kinda outside the realm of EVM execution, and more of in the domain of a "block execution" environment. Making this a Query would seem to result in storing all of the abi type information in the `result` of the VM as an intermediate failure, only to immediately resolve it with a fetcher. In the other cases, our `Query`s are at least something blocking the actual execution, so I think it doesn't feel too wild to allow `IOAct` here. But I'm down to experiment with removing `Query` entirely in a future pr

Reply:
yeah ok it's a fair point re: the beurocracy. I'm fine with this for now.